docs/en/platform-engineers/debug-test-cue.md

```diff
@@ -664,3 +664,465 @@ spec:
+upgrade an application.
+It basically generates a diff between the specific revision of an application 
+and the result of `vela system dry-run`.
+The result shows the changes (added/modified/removed/no_change) of the application as well as its sub-resources, such as components and traits.
+`live-diff` will not make any changes to the living cluster, so it's very
+helpful if you want to update an application but worry about the unknown results
+that may be produced.
+
+Let's prepare an application and the updated configuration with which we plan to
+update the application.
+
+```yaml
+# app.yaml
+apiVersion: core.oam.dev/v1beta1
+kind: Application
+metadata:
+  name: livediff-demo
+spec:
+  components:
+    - name: myweb-1
+      type: myworker
+      properties:
+        image: "busybox"
+        cmd:
+          - sleep
+          - "1000"
+        lives: "3"
+        enemies: "alien"
+      traits:
+        - type: myingress
+          properties:
+            domain: "www.example.com"
+            http:
+              "/": 80
+        - type: myscaler
+          properties:
+            replicas: 2
+    - name: myweb-2
+      type: myworker
+      properties:
+        image: "busybox"
+        cmd:
+          - sleep
+          - "1000"
+        lives: "3"
+        enemies: "alien"
+```
+
+```yaml
+# app-updated.yaml
+apiVersion: core.oam.dev/v1beta1
+kind: Application
+metadata:
+  name: livediff-demo
+spec:
+  components:
+    - name: myweb-1
+      type: myworker
+      properties:
+        image: "busybox"
+        cmd:
+          - sleep
+          - "2000" # change a component property
+        lives: "3"
+        enemies: "alien"
+      traits:
+        - type: myingress
+          properties:
+            domain: "www.example.com"
+            http:
+              "/": 90 # change a trait
+        # - type: myscaler # remove a trait
+        #   properties:
+        #     replicas: 2
+    - name: myweb-2
+      type: myworker
+      properties: # no change on component property
+        image: "busybox"
+        cmd:
+          - sleep
+          - "1000"
+        lives: "3"
+        enemies: "alien"
+      traits:
+        - type: myingress # add a trait
+          properties:
+            domain: "www.example.com"
+            http:
+              "/": 90
+    - name: myweb-3  # add a component
+      type: myworker
+      properties:
+        image: "busybox"
+        cmd:
+          - sleep
+          - "1000"
+        lives: "3"
+        enemies: "alien"
+      traits:
+        - type: myingress
+          properties:
+            domain: "www.example.com"
+            http:
+              "/": 90
+```
+ComponentDefinitions and TraitDefinitions used in this sample are stored in 
+`./doc/examples/live-diff/definitions`.
+
+Let's deploy the application.
+```shell
+kubectl apply ./doc/examples/live-diff/definitions
+kubectl apply ./doc/examples/live-diff/app.yaml
+```
```

Comment:
move this section before the `updated yaml`, create the application first and expalin we want to update like that

Reply:
ok